toolchain/check/testdata/class/no_prelude/name_poisoning.carbon

```diff
@@ -2,14 +2,244 @@
 // Exceptions. See /LICENSE for license information.
 // SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 //
-// EXTRA-ARGS: --no-dump-sem-ir
-//
 // AUTOUPDATE
 // TIP: To test this file alone, run:
 // TIP:   bazel test //toolchain/testing:file_test --test_arg=--file_tests=toolchain/check/testdata/class/no_prelude/name_poisoning.carbon
 // TIP: To dump output, run:
 // TIP:   bazel run //toolchain/testing:file_test -- --dump_output --file_tests=toolchain/check/testdata/class/no_prelude/name_poisoning.carbon
 
+// --- no_poison.carbon
+
+library "[[@TEST_NAME]]";
+
+class C;
+
+// N.F uses N.C and not C.
+namespace N;
+class N.C;
+fn N.F(x: C);
```

Comment:
Seems like there should be a call that verifies the truth of the comment, like:
```
fn TestCall(y: N.C) {
  // `N.F` accepts an `N.C` not a `package.C`.
  N.F(y);
}
```

Reply:
Done.
Note though that this is implicitly tested in the sem ir dump.

Reply:
I agree and I do see the value, though it does add some complexity.
Technically the test would fail if something goes wrong even if it only shows in the sem ir dump.